tsdb/engine/tsm1/cache.go

```diff
@@ -306,11 +324,24 @@ func (c *Cache) Values(key string) Values {
 
 // Delete will remove the keys from the cache
 func (c *Cache) Delete(keys []string) {
+	c.DeleteRange(keys, math.MinInt64, math.MaxInt64)
+}
+
+// DeleteRange will remove the keys containing points between min and max from the cache
+func (c *Cache) DeleteRange(keys []string, min, max int64) {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
 	for _, k := range keys {
-		delete(c.store, k)
+		origSize := c.store[k].size()
+		if min == math.MinInt64 && max == math.MaxInt64 {
+			c.size -= uint64(origSize)
+			delete(c.store, k)
+			continue
+		}
+
+		c.store[k].filter(min, max)
```

Comment:
We should probably check to see if all  the values were removed for this key here, and then delete it if it is now empty.